tools/webdriver/webdriver/bidi/modules/browsing_context.py

```diff
@@ -177,16 +182,16 @@ def _print(self, result: Mapping[str, Any]) -> Any:
     @command
     def set_viewport(self,
                      context: str,
-                     viewport: Optional[Mapping[str, Any]] = None,
-                     device_pixel_ratio: Optional[float] = None) -> Mapping[str, Any]:
+                     viewport: Union[Optional[Mapping[str, Any]], Undefined] = UNDEFINED,
+                     device_pixel_ratio: Union[Optional[float], Undefined] = UNDEFINED) -> Mapping[str, Any]:
         params: MutableMapping[str, Any] = {
             "context": context,
         }
 
-        if viewport is not None:
+        if viewport != UNDEFINED:
```

Comment:
It's a bit unfortunate that this changes the behavior for just these two arguments compared to any other BiDi command. I assume that we should have better named it `NULL` instead of `UNDEFINED` so that by default not using the argument (`None`) would still mean we do not send the argument, but with `NULL` we set `params["viewport"] to `None`. I assume we could do that with the `__str__` method?

Reply:
I think it would be more clear to map None to null (the default json serialization behaviour in python's json AFAIK) and UNDEFINED to undefined, and I think it is fine to update the test code incrementally.

Reply:
I would have called this `Missing` rather than `UNDEFINED` because it makes the intent clearer without assuming js semantics around `undefined` vs `null`. But in general having a different sentinel for missing values vs values that are present but have the JSON value `null` does make sense.

Reply:
Looks like we have consesus so lets leave as is for now. 

Btw. can't we use `if viewport is not UNDEFINED` here? It's a singleton so this should basically work, right?

Reply:
Done.